ethereum/core/src/main/java/org/hyperledger/besu/ethereum/MainnetBlockValidator.java

```diff
@@ -110,7 +110,9 @@ public BlockValidator.Result validateAndProcessBlock(
         shouldPersist ? maybeWorldState.get() : maybeWorldState.get().copy();
 
     final BlockProcessor.Result result = processBlock(context, worldState, block);
-    if (result.isFailed()) {
+    if (result.isFailed() && result.causedBy().isPresent()) {
```

Comment:
why does the fact that there is a causedBy make us pass the condition? shouldn't we check what has a causedBy value?

Reply:
Because if we have an exception during block _processing_ we want to return a include that as the cause of our _validation_ result.  If it is not present, we just continue with the old logic. Trying to keep this low impact till we can think it through better.